### DISCUSSION TITLE
Error in corrMat[tf, targets] : subscript out of bounds

### DISCUSSION THREAD

**Discussion by KabitaBaral1** [aertslab/SCENIC#251] [R]
Hi,

I ran into the error:
Error in corrMat[tf, targets] : subscript out of bounds

right in this step and it still persists even when I changed code from :  
tfModules_withCorr_byTF <- lapply(tfModules_byTF[tfs], function(tfGeneSets)
{
tf <- as.character(unique(tfGeneSets$TF))
targets <- as.character(tfGeneSets$Target)
cbind(tfGeneSets, corr=c(as.numeric(corrMat[tf,targets] > 0.03) - as.numeric(corrMat[tf,targets] < -0.03)))
})

to:
tfModules_withCorr_byTF <- lapply(tfModules_byTF[tfs], function(tfGeneSets)
{
tf <- unique(tfGeneSets$TF)
targets <- tfGeneSets$Target
targets <- as.character(tfGeneSets$Target)
cbind(tfGeneSets, corr=c(as.numeric(corrMat[tf,targets] > 0.03) - as.numeric(corrMat[tf,targets] < -0.03)))
})

as suggested. Any idea on what might be the reason behind it?

Thank you

**Comment by YogiBud** [comment]
Hi,
I ran into the same error before, and in my case it was related to running the SCENIC pipeline inside a loop. SCENIC is somewhat hard-coded and automatically creates its own output folders, intermediate files, and figures. When you run it in a loop without changing directories, each iteration can overwrite/reuse files from the previous run. As a result, SCENIC may end up reusing or reading files that do not correspond to the current dataset, which can lead to errors like above.
